src/actions/guides/database/testing.cr

```diff
@@ -19,18 +31,137 @@ class Guides::Database::Testing < GuideAction
+    our `Post` model has a `column title : String` and `column body : String`, then the box will have
+    a `title` method and `body` method.
+
+    These methods take an argument of the default value you want to set for that box.
+
+    ```crystal
+    class PostBox < Avram::Box
+      def initialize
+        title "Milk was a bad choice"
+        body "Every post created from this PostBox will default to these values"
+      end
+    end
+    ```
+
+    A `Post` will usually have more columns like `id`, `created_at`, and `updated_at`. We can omit
+    these since Avram will handle setting values on these for us. We can also omit any other fields
+    that are nilable by default.
+
+    ### Sequences
+
+    Your model may have a unique constraint on a field like a `Post` title, for example. In this case,
+    you can use the `sequence` method to auto-increment a number on to your default value ensuring it will
+    be unique.
+
+    ```crystal
+    class PostBox < Avram::Box
+      def initialize
+        title sequence("My new blog post")
+      end
+    end
+    ```
+
+    The first time a `PostBox` is created, the `title` will be set to `"My new blog post-1"`. The next time
+    one is created, the `title` will be set to `"My new blog post-2"`, and so on.
+
+    > Sequences always return a `String`. For sequence type values on other types, you'll need to
+    > implement those yourself.
+
+    ## Associations with boxes
+
+    When you create a Box, you may want to have an association already set for you. In this case,
+    you'll just use that association's box to set the `foreign_key` value.
+
+    ```crystal
+    class PostBox < Avram::Box
+      def initialize
+        title sequence("post-title")
+        body "blah"
+        user_id UserBox.create.id
+      end
+    end
+    ```
+
+    > Creating a Box returns an instance of that model, which means you have access to all of the model
+    > methods.
+
     ## Saving records
 
+    As shown previously, a Box has a `create` method which saves the record to your database. A Box is
+    essentially a fancy wrapper around [SaveOperation](#{Guides::Database::ValidatingSavingDeleting.path}).
+
+    Boxes give you access to two helpful class methods `create` and `create_pair`.
+
+    ### Box.create
+
+    This will create a `Post`, and store it in your database, then return the instance of `Post`.
+    It will use the defaults you defined in your `initialize` method of `PostBox`.
+
     ```crystal
-    PostBox.create
-    PostBox.create_pair
+    post = PostBox.create
+    post.title #=> "post-title-1"
+
+    post2 = PostBox.create
+    post2.title #=> "post-title-2"
     ```
 
     ### Overriding data
 
+    When you need to override the defaults previously set, you'll pass a block to `create`.
+
+    ```crystal
+    PostBox.create do |box|
+      box.title("Draft")
+      box.body("custom body")
+    end
```

Comment:
For some reason it never clicked that I could use do/end. For longer stuff this is really nice!  So glad you  added that example

Reply:
Though maybe call the yielded var `post`? Up to you

Reply:
I thought about that, but you could define a custom method on the box that you'd have access to here, so I figured it's better to show that you're dealing with box here.

Reply:
Good point. I definitely like “box” better now!